### PR TITLE
AB#134108: Summary table loads when using novel compounds

### PR DESCRIPTION
**Pull request by JoeDavies-6**
# Overview

-Fixes a bug where the summary table didn't load when using novel compounds.
-Code to get the hazard data for the summary table has been moved into services and refactored into multiple functions
-Changes to JS files to add more JSDocs and fix vulnerabilities 

## Azure Boards

- [AB#134108](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/134108)

### CODE REVIEW

File: Webapp/sources/services/queries/hazards.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        a list of hazard codes with each h code as an item in the list
+
+    """
+    compound_hazard_codes = replace_substrings(
+        compound_hazard_codes, {"   ": ",", "-": ",", " + ": ",", "+": ","}
+    )
+    return sorted(list(set(compound_hazard_codes.split(","))))
+
+
+def get_hazardcode_database_object(h_code: str) -> models.HazardCode:
```
Comment (AndyRae) [inline Webapp/sources/services/queries/hazards.py:242]:
```suggestion
def get(h_code: str) -> models.HazardCode:
```

File: Webapp/sources/services/queries/hazards.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+    Returns:
+        str: The input string with substrings replaced according to the dictionary.
+    """
+    for key, value in replace_dict.items():
+        input_string = input_string.replace(key, value)
+    return input_string
+
+
+def get_data_from_h_code(h_code: str) -> Tuple[str, str]:
```
Comment (AndyRae) [inline Webapp/sources/services/queries/hazards.py:206]:
```suggestion
def get_data(h_code: str) -> Tuple[str, str]:
```

File: Webapp/sources/services/queries/compound.py [hunk 3]
```diff
@@ -23,3 +23,14 @@ def get_compound_data_error_reports() -> List[models.CompoundDataErrorReport]:
     """
     # Add time cut off if list grows too large
     return db.session.query(models.CompoundDataErrorReport).all()
+
+
+def get_smiles_from_primary_key(primary_key: int) -> str:
```
Comment (AndyRae) [inline Webapp/sources/services/queries/compound.py:28]:
```suggestion
def get_smiles(primary_key: int) -> str:
```

**Review by AndyRae — DISMISSED** [review]

File: Webapp/sources/services/queries/compound.py (hunk 3 above)
Comment (JoeDavies-6) [reply Webapp/sources/services/queries/compound.py:28]: Cheers Andy, much appreciated for these suggested changes!

**Review by AndyRae — APPROVED** [review]

**Review by blargly — APPROVED** [review]

**Pull request merged**